CodeChat_Server/windows_installer.ps1

```diff
@@ -0,0 +1,205 @@
+        (Be sure to add to the PATH)"
+        exit
+    }
+}
+# Returning to User Dir
+cd $env:USERPROFILE
+# |
+# 
+# Grabbing the version of python and splitting it up so as to make parsing easier
+$pythonVersion = python --version
+$pythonVersion = $pythonVersion.Split()[1]
+$pythonVersionMaj = $pythonVersion.Split('.')[0]
+$pythonVersionMin = $pythonVersion.Split('.')[1]
+
+# Version Checking
+# if the version we have is not greater than or equal to the required version, an updated version must be installed
+# checking negative case so that if it evaluates as true (version is less than required) then throw warning
+$InstallBool = False;
+if ($pythonVersionMaj -lt $pythonVersionReqArray[0]){
+    $InstallBool = True
+}
+elseif($pythonVersionMaj -eq $pythonVersionReqArray[0]){
+    if($pythonVersionMaj -lt $pythonVersionReqArray[1]){
+        $InstallBool = True
+    }
+}
+
+if($InstallBool){
+    $response = Read-Host -Prompt "You do not have the correct version of Python installed, would you like the script to install the required version ($pythonVersionReq) (y/n)?"
+    if($response -eq "y"){
+        echo "Downloading Python Version $PyTarget"
+        # Downloading Python Unsure if ``-UseDefaulCredential`` in necessary
+        Invoke-WebRequest -Uri $PythonURL -OutFile $PythonDest -UseDefaultCredential
+        echo "Python Version$PyTarget Downloaded
+        
+        Installing Python"
+        Read-Host "Press any button to continue, then click 'Install Now' in popup"
+        cd $env:USERPROFILE\Downloads
+
+        # Installs Python ``Out-Null`` option makes sure the program doesn't continue before the installation completes Using Start-Process over just using the cmd because it allows me to input a string, so we can easily change the target python version. 
+        Start-Process ".\python-$PyTarget-amd64.exe" -ArgumentList /quietly,PrependPath=1 -Wait
+
+        echo "Python Version $PyTarget Installed Successfully
+        "
+    }
+    else{
+        echo "Please install the required version of python: $pythonVersionReq or later
+        "
+        exit
+    }
+}
+else{
+    echo "Acceptable Version of Python found:"
+}
+
+# Pip Installation
+# ----------------
+# Make sure pip, the Python installer, is up to date on Windows
+echo "Upgrading pip..."
+python -m pip install --upgrade pip
+echo "
+"
+
+# Going to the user dir
+cd $env:USERPROFILE
+
+# Creating codechat venv
+# ----------------------
+# We do this so that our installation does not mess with any other installations of python
+echo "Creating codechat venv..."
+$codechat = Get-Item codechat
+if([string]::IsNullOrEmpty($codechat)){
+    # Create a virtual enviroment named codechat
+    python -m venv codechat
+    echo "Virtual Enviroment Created Sucessfully
+    "
+}
+else{
+    echo "'codechat' virtual enviroment already found, skipping this step
+    "
+}
+
+echo "Activating Virutal enviroment"
```

Comment:
Wrong approach -- don't activate the venv, just run (I think) `codechat\scripts\python -m pip install CodeChat-System`.

Reply:
should we still install/upgrade pip earlier? 

`
# Pip Installation
# ----------------
# Make sure pip, the Python installer, is up to date on Windows
echo "Upgrading pip..."
python -m pip install --upgrade pip
echo " "
`
or should we check right before we install after we are in the virtual environment e.g. 
`\codechat\Scripts\python -m pip install --upgrade pip`

Reply:
Run this inside the venv instead.

Reply:
1. commented out activating the venv and everything worked.
2. added `\codechat\Scripts\` to the install commands and: 
a. powershell script gave blank codechat path
b. can't run codechat in VSC (can't find exe)

Reply:
removed the backslash before `codechat\Scripts\` and now it is behaving properly